htmltools/_core.py

```diff
@@ -561,6 +562,8 @@ def __init__(
         kids = [x for x in args if not isinstance(x, dict)]
         self.children = TagList(*kids)
 
+        self.displayhook_stack: list[Callable[[object], None]] = []
```

Comment:
Am I right that this stack can have at most one item? If I understand correctly, the "stackiness" is actually provided by Python calling the `__enter__` methods of one context manager after another.

If that's correct, then I suggest changing this to:
```suggestion
        self.prev_displayhook: Callable[[object], None] | None = None
```

Reply:
I think I thought there was a subtle edge case where you'd need to keep track of multiple but now I can't think of what that would be. Nested use of one tag object? That gives you infinite recursion (but we should still make sure it doesn't break the pre-existing displayhook state)

Reply:
Maybe better to throw if `__enter__` is called and `prev_displayhook` is not `None`